tests/test_node.py

```diff
@@ -219,3 +241,64 @@ def test_clear():
         assert child.root == child
 
     assert node.children == []
+
+
+def test_style_and_applicator_assignment():
```

Comment:
These should be more self-contained tests of specific features/API usage. Having more small tests makes it easier to work out the exact cause of a problem.